chore(wallet)_: fee handling improvements for Status Network Sepolia

saledjenic opened the pull request:
- Added L1 fee for the Base chain, since it is based on Optimism stack and should follow the same fee calculation.
- Remove unused functions `HasNoBaseFee` and `HasNoPriorityFee` from wallet utilities since that should not be hardcoded, but dynamically evaluated.
- A temporary patch was added to the `getEIP1559SuggestedFees` function to handle the specificities of the Status chain.

The last commit bring a dynamic fee evaluation for Status Sepolia chain:
- Added support for the `linea_estimateGas` method in the wallet's gas estimation logic.
- Updated the `EstimateGas` function to route to `linea_estimateGas` when the chain ID is for Status Sepolia chain.
- Enhanced fee management to dynamically evaluate the account that is going to place a tx is a base fee less
and/or priority fee less for the Status Sepolia chain.

saledjenic committed:
fix(wallet)_: add l1 fee to a base chain transactions

The Base chain is based on Optimism stack, so should follow the same fee calculation.

saledjenic committed:
chore(wallet)_: fee handling improvements for Status Network Sepolia

- Remove unused functions `HasNoBaseFee` and `HasNoPriorityFee` from wallet utilities since that should not be hardcoded, but dynamically evaluated.
- A temporary patch was added to the `getEIP1559SuggestedFees` function to handle the specificities of the Status chain.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (80)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/1/) | 2025-06-25 13:38:08 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250625-133456-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/1/) | 2025-06-25 13:38:19 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250625-133448-7d9d25-pr6710.aar) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/1/) | 2025-06-25 13:39:49 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250625-133503-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/1/) | 2025-06-25 13:39:58 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250625-133450-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/1/) | 2025-06-25 13:41:13 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250625-133458-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/1/) | 2025-06-25 13:41:46 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250625-133452-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/1/) | 2025-06-25 13:44:49 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/1//consoleText) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/1/) | 2025-06-25 13:46:12 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250625-133453-7d9d25-pr6710.zip) |
| :heavy_check_mark: | 7d9d25cc | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/1/) | 2025-06-25 14:05:26 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/2/) | 2025-06-26 14:26:32 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250626-142311-305cdb-pr6710.aar) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/2/) | 2025-06-26 14:26:57 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250626-142316-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/2/) | 2025-06-26 14:27:16 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250626-142312-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/2/) | 2025-06-26 14:28:43 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250626-142319-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/2/) | 2025-06-26 14:29:16 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250626-142323-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/2/) | 2025-06-26 14:29:27 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/2//consoleText) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/2/) | 2025-06-26 14:29:59 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250626-142311-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/2/) | 2025-06-26 14:33:55 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250626-142314-305cdb-pr6710.zip) |
| :heavy_check_mark: | 305cdb52 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/2/) | 2025-06-26 14:48:24 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/3/) | 2025-07-01 16:09:38 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250701-160612-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/3/) | 2025-07-01 16:10:24 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250701-160607-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/3/) | 2025-07-01 16:10:52 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250701-160605-fecfc8-pr6710.aar) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/3/) | 2025-07-01 16:10:58 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250701-160609-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/3/) | 2025-07-01 16:11:13 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250701-160615-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/3/) | 2025-07-01 16:13:11 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250701-160620-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/3/) | 2025-07-01 16:13:49 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/3//consoleText) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/3/) | 2025-07-01 16:19:03 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250701-160609-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/3/) | 2025-07-01 16:35:02 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/3//consoleText) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/4/) | 2025-07-03 07:56:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250703-075438-fecfc8-pr6710.aar) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/4/) | 2025-07-03 07:57:55 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250703-075441-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/4/) | 2025-07-03 07:58:06 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250703-075448-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/4/) | 2025-07-03 07:58:21 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250703-075445-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/4/) | 2025-07-03 07:58:53 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250703-075440-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/4/) | 2025-07-03 08:00:20 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250703-075451-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/4/) | 2025-07-03 08:00:53 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/4//consoleText) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/4/) | 2025-07-03 08:06:14 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250703-075442-fecfc8-pr6710.zip) |
| :heavy_check_mark: | fecfc8d8 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/4/) | 2025-07-03 08:20:58 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/5/) | 2025-07-03 08:00:42 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250703-075831-5670e9-pr6710.aar) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/5/) | 2025-07-03 08:01:08 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250703-075834-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/5/) | 2025-07-03 08:01:50 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250703-075838-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/5/) | 2025-07-03 08:03:15 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250703-075901-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/5/) | 2025-07-03 08:03:36 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250703-075841-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/5/) | 2025-07-03 08:05:36 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250703-080022-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/5/) | 2025-07-03 08:07:07 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/5//consoleText) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/5/) | 2025-07-03 08:15:54 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250703-080619-5670e9-pr6710.zip) |
| :heavy_check_mark: | 5670e91c | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/5/) | 2025-07-03 08:46:29 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/6/) | 2025-07-04 11:35:57 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250704-113340-cb6372-pr6710.aar) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/6/) | 2025-07-04 11:36:58 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250704-113345-cb6372-pr6710.zip) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/6/) | 2025-07-04 11:37:04 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250704-113343-cb6372-pr6710.zip) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/6/) | 2025-07-04 11:37:59 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250704-113341-cb6372-pr6710.zip) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/6/) | 2025-07-04 11:38:47 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250704-113348-cb6372-pr6710.zip) |
| :heavy_multiplication_x: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/6/) | 2025-07-04 11:39:18 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/6//consoleText) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/6/) | 2025-07-04 11:39:33 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250704-113352-cb6372-pr6710.zip) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/6/) | 2025-07-04 11:43:18 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250704-113343-cb6372-pr6710.zip) |
| :heavy_check_mark: | cb637209 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/6/) | 2025-07-04 12:00:57 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/6//consoleText) |
| | | | | | | |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/7/) | 2025-07-04 11:50:44 | ~51 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/7/) | 2025-07-04 11:50:49 | ~51 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/7/) | 2025-07-04 11:50:54 | ~56 sec | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/7/) | 2025-07-04 11:51:14 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/7/) | 2025-07-04 11:51:26 | ~1 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/7/) | 2025-07-04 11:52:31 | ~2 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/7/consoleText) |
| :x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/7/) | 2025-07-04 11:53:11 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/7/consoleText) |
| :heavy_multiplication_x: | d87ba4b9 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/7/) | 2025-07-04 11:54:58 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/7//consoleText) |
| | | | | | | |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/8/) | 2025-07-04 11:52:34 | ~48 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/8/) | 2025-07-04 11:52:39 | ~49 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/8/) | 2025-07-04 11:52:45 | ~55 sec | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/8/) | 2025-07-04 11:52:57 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/8/) | 2025-07-04 11:53:15 | ~1 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/8/) | 2025-07-04 11:55:09 | ~2 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/8/consoleText) |
| :x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/8/) | 2025-07-04 11:56:23 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/8/consoleText) |
| :heavy_multiplication_x: | 83ceb964 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/8/) | 2025-07-04 11:57:09 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/8//consoleText) |
| :heavy_multiplication_x: | 83ceb964 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/7/) | 2025-07-04 12:01:59 | ~55 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/7//consoleText) |
| | | | | | | |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/9/) | 2025-07-04 12:37:17 | ~48 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/9/consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/9/) | 2025-07-04 12:37:22 | ~50 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/9/consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/9/) | 2025-07-04 12:37:28 | ~55 sec | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/9/consoleText) |
| :heavy_multiplication_x: | 9e7c21ce | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/8/) | 2025-07-04 12:37:31 | ~53 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/8//consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/9/) | 2025-07-04 12:37:49 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/9/consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/9/) | 2025-07-04 12:38:00 | ~1 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/9/consoleText) |
| :heavy_multiplication_x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/9/) | 2025-07-04 12:38:32 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/9//consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/9/) | 2025-07-04 12:39:07 | ~2 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/9/consoleText) |
| :x: | 9e7c21ce | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/9/) | 2025-07-04 12:39:29 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/9/consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/10/) | 2025-07-04 12:58:55 | ~51 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/10/consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/10/) | 2025-07-04 12:59:03 | ~56 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/10/consoleText) |
| :heavy_multiplication_x: | ed444557 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/9/) | 2025-07-04 12:59:09 | ~55 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/9//consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/10/) | 2025-07-04 12:59:18 | ~1 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/10/consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/10/) | 2025-07-04 12:59:39 | ~1 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/10/consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/10/) | 2025-07-04 12:59:48 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/10/consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/10/) | 2025-07-04 13:00:49 | ~2 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/10/consoleText) |
| :x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/10/) | 2025-07-04 13:01:17 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/10/consoleText) |
| :heavy_multiplication_x: | ed444557 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/10/) | 2025-07-04 13:02:44 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6710/11/) | 2025-07-04 13:21:39 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250704-131855-34394d-pr6710.aar) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6710/11/) | 2025-07-04 13:21:53 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250704-131858-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6710/11/) | 2025-07-04 13:22:10 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250704-131902-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6710/11/) | 2025-07-04 13:23:11 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250704-131857-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6710/11/) | 2025-07-04 13:24:01 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250704-131904-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/11/) | 2025-07-04 13:24:25 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6710/11//consoleText) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6710/11/) | 2025-07-04 13:24:38 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250704-131909-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6710/11/) | 2025-07-04 13:29:04 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250704-131859-34394d-pr6710.zip) |
| :heavy_check_mark: | 34394dd6 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/10/) | 2025-07-04 13:45:27 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6710/10//consoleText) |

saledjenic committed:
feat(wallet)_: implement linea_estimateGas for Status Sepolia chain

- Added support for the `linea_estimateGas` method in the wallet's gas estimation logic.
- Updated the `EstimateGas` function to route to `linea_estimateGas` when the chain ID is for Status Sepolia chain.
- Enhanced fee management to dynamically evaluate the account that is going to place a tx is base fee less
and/or priority fee less for the Status Sepolia chain.

vkjr commented on services/wallet/router/fees/fees.go:
```diff
@@ -110,8 +111,10 @@ type FeeManager struct {
-	if common.IsGaslessChainAndEIP1559Compatible(chainID) {
-		return true, nil
+	// Since the Status Network is gasless chain, but EIP-1559 compatible, we should not rely on checking the BaseFeePerGas, that's why we have this special case.
+	eip1559Enabled, err := common.IsPartiallyOrFullyGaslessChainEIP1559Compatible(chainID)
```
`IsPartiallyOrFullyGaslessChainEIP1559Compatible` function looks like overcomplication for me. If all we want to check is that chain is EIP1559-compatible, we don't have to bring the concept of "gasless" into play. We can have function `common. IsKnownEIP1559CompatibleChain` and use it instead and put the check for `status` chain there.

saledjenic replied:
@vkjr not really. I prefer naming things what they truly are and what they truly do.

>  If all we want to check is that chain is EIP1559-compatible

Yes, the function `IsEIP1559Enabled` really checks if a chain X is EIP1559 compatible or not, and it doesn't bring "gasless" into play, and works for any chain. But the logic for checking that for gasless chains and partially gasless chains differs from checking that for other chains.

> We can have function common. IsKnownEIP1559CompatibleChain

Why do we need that one if we already have `IsEIP1559Enabled` that does that?

vkjr replied:
>Why do we need that one if we already have IsEIP1559Enabled that does that?

Well, right now `IsEIP1559Enabled` calls `IsPartiallyOrFullyGaslessChainEIP1559Compatible` which calls `!IsPartiallyOrFullyGaslessChain` and all these nested calls do is check if `chainId` is Status Sepolia. Just putting this check inside `isEIP1559Compatible` would make things more readable.

vkjr replied:
Buy saying "bring the concept of "gasless" into play" I mean that the word "gasless" appears inside the `IsEIP1559Enabled` function when we call `IsPartiallyOrFullyGaslessChainEIP1559Compatible`.

So for someone like me who is less aware of a codebase and reads the code of `IsEIP1559Enabled` that poses a question "what gasless have to do with EIP 1559", so I'm talking more about cognitive load of reading the code here :)

saledjenic replied:
Even it's a simple function:
```go
func IsPartiallyOrFullyGaslessChain(chainID uint64) bool {
	return chainID == StatusNetworkSepolia
}
```
it says a lot. Has a meaningful name, is used in more than a single place in the code, and having in mind that tomorrow when we need to add StatusMainnetChain (or maybe some other gasless chain) that will be just a single line change in that function, it deserves to be a separate function.

saledjenic replied:
> So for someone like me who is less aware of a codebase and reads the code of IsEIP1559Enabled that poses a question "what gasless have to do with EIP 1559"...

So for any dev who intends to use the function will see `IsEIP1559Enabled` and without looking at the implementation should be safe to call it for any chain and it will work. 

If one would like to call `IsPartiallyOrFullyGaslessChainEIP1559Compatible`, there is a function description  that says `// IsPartiallyOrFullyGaslessChainEIP1559Compatible throws an error if the chain is not partially or fully gasless, if it is, returns true if the chain is EIP-1559 compatible` so since the check is the same either the chain is fully gasless or partially gasless (has priority fees) my intention was to be explicit in function naming, and agree that maybe it looks cumbersome but has the meaning.

saledjenic replied:
@vkjr the point is that checking for the Status chain should not be at all in `getEIP1559SuggestedFees`, I've added it there temporarily (cause some non-zero values were returned for the priority fees for `eth_feeHistory` and that was wrong) so that's just a patch that is going to be removed (see the comment there).

vkjr replied:
@saledjenic, sorry, I move my initial comment to other thread thinking it will be more logical but you already answered it and now this conversation is harder to grasp, my bad. [this is the message](https://github.com/status-im/status-go/pull/6710/files#r2179959436) that you answered and that I moved to other thred.

Well, `getEIP1559SuggestedFees` named like an appropriate place to get fees for any chain. And even if it will change someday doesn't mean we shouldn't make it clean now. Because all my arguments remain - `noBaseFee` and `noPriorityFee` logically doesn't belong to SuggestedFees.

saledjenic replied:
@vkjr this is the PR that I've tested, I don't have anything against it if you're willing to make such changes, but let's merge this PR and feel free to improve it after that.

vkjr replied:
@saledjenic, but what is the point of reviewing PRs then? :) 
I fully understand the desire to merge because I also have it for every PR I've created. But reviewer also bears a responsibility to ensure that code is correct and maintainable. And I think that in its current state the code is harder to understand and maintain and it shouldn't be merged "as is". The time that we will gain by merging it now is a debt that will be paid by future code readers when they try to understand these parts.

saledjenic replied:
> but what is the point of reviewing PRs then? :)

@vkjr yes, the point is to make the PR code better in all senses, to ensure quality, security, correctness and so on... but we're discussing for two days a fully correct code or I don't understand your point, what you really want? :)

Apart from other things (those that I already commented on why they are as they are, and why we don't need to change them), will sum up the remaining that we're discussing:
- You asked to merge `getEIP1559SuggestedFees` and `f.getGaslessParamsForAccount` but I explained why that's not needed since the condition for the Status chain will go away soon, that's a temporary thing cause we had an issue on the Status chain.
- You said "logically doesn't belong to SuggestedFees" but not sure where did you see that they belong to `SuggestedFees`? :) `noBaseFee` and `noPriorityFee` do not belong to `SuggestedFees`, but they are used to calculate `SuggestedFees`, so what's wrong in that?

Feel free to write it as you think it should be, just curious about your solution.

vkjr replied:
@saledjenic, there are few things that bother me in the code
- same comparison `if chainID == common.StatusNetworkSepolia` in nested functions: `SuggestedFees` and `getEIP1559SuggestedFees`. 
- `noBaseFee` and `noPriorityFee` params that are calculated in `SuggestedFees` but really used only in `r.evaluateAndUpdatePathDetails`. I disagree with statement that they are also used in calculating fees because as I mentioned in [comment](https://github.com/status-im/status-go/pull/6710#discussion_r2179959436), `baseFee` is always 0 or nil independntly from the result of `getGaslessParamsForAccount` call.

So my suggestion is to move `getGaslessParamsForAccount` call out of `SuggestedFees`.

In this case we will get following:

router.go: gets `noBaseFee` and `noPriorityFee` on its own
```go
func (r *Router) setCustomTxDetails(ctx context.Context, pathTxIdentity *requests.PathTxIdentity, pathTxCustomParams *requests.PathTxCustomParams) error {
	...
	fetchedFees, err := r.feesManager.SuggestedFees(ctx, pathTxIdentity.ChainID, addrFrom)}
	noBaseFee, noPriorityFee, err := r.feesManager.getGaslessParamsForAccount(ctx, pathTxIdentity.ChainID, addrFrom) // <- moved here
	for _, path := range r.activeRoutes.Route {
	...
		err = r.evaluateAndUpdatePathDetails(ctx, path, fetchedFees, usedNonces, noBaseFee, noPriorityFee, false, 0)
		...	
	}
```

fees.go:
```go
setCustomTxDetailsfunc (f *FeeManager) SuggestedFees(ctx context.Context, chainID uint64, address ethCommon.Address) (suggestedFees *SuggestedFees, err error) {

	// check `if chainID == common.StatusNetworkSepolia`
	// is not needed here because `getEIP1559SuggestedFees` checks it already
	// and zeroes-out all the fees

	lowPriorityFeePerGasLowerBound, mediumPriorityFeePerGas, maxPriorityFeePerGasUpperBound, baseFee, err = getEIP1559SuggestedFees(chainID, feeHistory)
		if err != nil {
			suggestedFees, err = f.getNonEIP1559SuggestedFees(ctx, chainID)
			return
		}
	...
}
```

This way we don't need `noBaseFee` and `noPriorityFee` inside `SuggestedFees` and also do not need to check for Status Sepolia network.
Wdyt?

saledjenic replied:
> same comparison if chainID == common.StatusNetworkSepolia in nested functions: SuggestedFees and getEIP1559SuggestedFees.

@vkjr I have a feeling that you're not reading what I'm saying. That condition in `getEIP1559SuggestedFees` will go away soon, said that a few times already. Don't look at it.

saledjenic replied:
Just because of the discussion that you ran here, which I wanted to avoid, and don't waste time, I added this comment 
```
// This is something we can do just now (cause Lenea call has some issues), but should not be used later (after September 2025),
// instead `linea_estimateGas` call should be used for the Status Network.
``` 
at the moment when I added that condition there.

vkjr replied:
@saledjenic, I've heard that this condition will go away "soon". And I don't see how this as a reason for not improving code "now". Because we all know how delayed can be expected changes. 

Also, even if initialization to zeros will go away, where exactly will reside the receiving of actual values with `linea_estimateGas`? Right now I see 2 functions within `SuggestedFees`: `getEIP1559SuggestedFees` and `getNonEIP1559SuggestedFees`. Receiving fees won't be in any of them? 

Speaking of hearing each other. Could you please comment on the whole point of having `if chainID == common.StatusNetworkSepolia` check within `SuggestedFees`. As I noted few times, all it does is initializing nil values to zeros. And we do not treat those values any different. What is the point?

saledjenic replied:
> And I don't see how this as a reason for not improving code "now".

Cause that's a temporary code, it doesn't hurt, but even if we don't change it in 3 months, but in 6 months for example, it will boil down to 3 lines (condition) removal and that's it, the rest of the code is untouched and clean. Also that's the point of the comment there. 

> Right now I see 2 functions within SuggestedFees: getEIP1559SuggestedFees and getNonEIP1559SuggestedFees. Receiving fees won't be in any of them?

Ofc not, they receive `feeHistory` as an input param and do the calculation.

> Could you please comment on the whole point of having if chainID == common.StatusNetworkSepolia check within SuggestedFees. 

There are only 2 places where we have `if chainID == common.StatusNetworkSepolia`, and those are:
1. in `SuggestedFees`
2. in `getEIP1559SuggestedFees`

We ignore the second one cause it's temporary here and will go away soon. That's just a patch for the Status chain. As soon as the fix lands in the Status chain, we can remove it.

In the first one remains to differentiate Status chain from other chains cause it fetches details from `linea_estimateGas`, to be clear, the condition could go to `getEIP1559SuggestedFees`, but then the function would need to be a member of `FeeManager`.

vkjr replied:
>There are only 2 places where we have if chainID == common.StatusNetworkSepolia, and those are:

> 1. in SuggestedFees
> 2. in getEIP1559SuggestedFees

>We ignore the second one cause it's temporary here and will go away soon. That's just a patch for the Status chain. As soon as the fix lands in the Status chain, we can remove it.

I'm fully okay with this one, I think it fully covers initialization needs for now.

>In the first one remains to differentiate Status chain from other chains cause it fetches details from linea_estimateGas, to be clear, the condition could go to getEIP1559SuggestedFees, but then the function would need to be a member of FeeManager.

This one I think should be removed because at the moment we don't have `linea_estimateGas` and the whole `if` condition doesn't make anything functional. Receiving `noBaseFee` and `noPriorityFee` within it also has no practical sense, it only pollutes `SuggestedFees` returned values and can be done outside of `SuggestedFees`.

saledjenic replied:
> This one I think should be removed because at the moment we don't have linea_estimateGas 

How do you mean that? It must stay and we have to use `linea_estimateGas`. The only question is, will we keep the call in `SuggestedFees` or move it to `getEIP1559SuggestedFees` and make that function a member of `FeeManager`. Those are only 2 possible options and no other way.

vkjr commented on services/wallet/router/fees/fees_history.go:
```diff
@@ -47,9 +50,36 @@ func (f *FeeManager) getFeeHistory(ctx context.Context, chainID uint64, newestBl
 
+func (f *FeeManager) getGaslessParamsForAccount(ctx context.Context, chainID uint64, address ethCommon.Address) (noBaseFee bool, noPriorityFee bool, err error) {
+	if !walletCommon.IsPartiallyOrFullyGaslessChain(chainID) {
+		return false, false, nil
```
Should this be an error condition?

saledjenic replied:
No, it doesn't. Why do you think so?

vkjr replied:
I noticed that we call this function on the top level under condition `chainID == common.StatusNetworkSepolia`, so it looked like it is intended to be called for gasless chains.

saledjenic replied:
Yes we call it under that condition, but that's our need in `SuggestedFees` function, but this function works for any chain supported in the app, not only for StatusNetworkSepolia chain.

vkjr replied:
Okay, in this function we do check `IsPartiallyOrFullyGaslessChain` but it should work for any network, got it 👍 

At the same time in `SuggestedFees` we firstly check if chainId is status sepolia to initialize fees to zeros. I have a feeling that the same logic should work for any gasless chain, not only for status sepolia.

So it probably would be more logical to cal the `getGaslessParamsForAccount` first and then depending on returned values of `noBaseFee` and `noPriorityFee` initialize fee values with zeros or with the result of `getEIP1559SuggestedFees`. In this way we will avoid manual check for sepolia status chain id and will rely on  more generic `IsPartiallyOrFullyGaslessChain` used within `getGaslessParamsForAccount`.
Wdyt?

saledjenic replied:
Yes, I agree, but isn't this more optimised :) like in that case call to `getGaslessParamsForAccount` and the code between `145` and `159` lines would be done for every chain, instead of as it's now, that we call that function only if it's StatusSepoliaChain. In all cases we need `feeHistory`, but `getGaslessParamsForAccount` are needed only in case of StautsSepoliaChain.

vkjr replied:
I lean toward the opinion that preliminary optimization is evil and personally always prefer code simplicity unless cpu or memory profiler said I'm wrong)

But I see your point, that also doesn't look beautiful and I think now I may understand why.. because we actually already have a function that should compute all fees - `getEIP1559SuggestedFees`. This function inside already has a check for status sepolia (line 79)! And on top of it we do this check again and set some fees to zero in `SuggestedFees.

I think we may merge `getEIP1559SuggestedFees` and `f.getGaslessParamsForAccount` which will make the `SuggestedFees` much cleaner. 
Also `getEIP1559SuggestedFees` can return the whole populated `suggestedFees` structure, it has all data to do this.

wdyt?

saledjenic replied:
~Yes, that's cleaner.~

@vkjr actually not, that's not good, cause everything done in this PR has its own reason why it's like that and I would keep it as it is. We cannot merge it cause moving `getGaslessParamsForAccount` into `getEIP1559SuggestedFees` will cause other not that nice changes, `getGaslessParamsForAccount` is a member of `FeeManager`, then we would need to pass context and address to `getEIP1559SuggestedFees` which is needless and also acting on the error returned by is is not the same. Those are different cases. So no changes on this topic.

vkjr replied:
Looks like I initially posted this in a wrong theread. Comment should be in this one.


@saledjenic, could you please help me understand the difference in these few cases, I'm not sure I correctly grasp them, maybe because of lack of Go knowledge.

Let's say we are in `SuggestedFees`.
We have `baseFee *big.Int` (which will be inialized to `nil` at first)

**Case 1**: Network is Status Sepolia and `noBaseFee` returned from `f.getGaslessParamsForAccount` equals true.
In this case we will get following sequence:
- we are entering `if chainID == common.StatusNetworkSepolia`
- `noBaseFee` equals true
- `baseFee = big.NewInt(0)` //doing conditional initialization
- we are exiting `if` block 
- `BaseFee: baseFee` //suggestedFees.BaseFee initialized with 0

**Case 2**: Network is Status Sepolia and `noBaseFee` returned from `f.getGaslessParamsForAccount` equals false.
- we are entering `if chainID == common.StatusNetworkSepolia`
- `noBaseFee` equals false
- we are exiting `if` block 
- `BaseFee: baseFee` //suggestedFees.BaseFee will remain `nil`

**Case 3, Hypothetical**: Let's say network is Status Sepolia but we don't have a call to `f.getGaslessParamsForAccount` in code and only call `f.getEIP1559SuggestedFees`. In this case:
- we enter `getEIP1559SuggestedFees`
- `if chainID == common.StatusNetworkSepolia {` //condition meets
- `return big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil` // we return 0 values
- exit `getEIP1559SuggestedFees`
- `BaseFee: baseFee` //suggestedFees.BaseFee initialized with 0


As we can see all these 3 cases leave us with the same result as for fees. Base fee will be 0 or `nil` which is indifferent in this case (correct me if I'm wrong) because we are never check for `nil` later.

So that means that call to `f.getGaslessParamsForAccount` has no practical value for calculating fees.
As you have said earlier,`noBaseFee` and `noPriorityFee` are the parameters of chain. 

So instead of calling `f.getGaslessParamsForAccount` inside `SuggestedFees` (which doesn't affect fees at all) and dragging two fee-unrelated params out of `SuggestedFees` to `r.evaluateAndUpdatePathDetails`, we can get rid of that call inside `SuggestedFees` at all and rely only on what `f.getEIP1559SuggestedFees` returns, while the call to `f.getGaslessParamsForAccount` can be moved to `evaluateAndUpdatePathDetails`. Or maybe it is not needed inside `evaluateAndUpdatePathDetails` but can be placed before it, on the same level as `SuggestedFees`.

What we will get:
- no double check for sepolia status in `SuggestedFees` and in `f.getEIP1559SuggestedFees`
- no need to initialize `&SuggestedFees` structure  in `SuggestedFees` func, it can be done inside `f.getEIP1559SuggestedFees`
- no dragging parameters 

I think that would be much cleaner, wdyt?

vkjr commented on services/wallet/router/fees/fees.go:
```diff
@@ -125,18 +128,44 @@ func (f *FeeManager) IsEIP1559Enabled(ctx context.Context, chainID uint64) (bool
 }
 
-func (f *FeeManager) SuggestedFees(ctx context.Context, chainID uint64) (*SuggestedFees, error) {
+func (f *FeeManager) SuggestedFees(ctx context.Context, chainID uint64, address ethCommon.Address) (suggestedFees *SuggestedFees, noBaseFee bool, noPriorityFee bool, err error) {
```
`noBaseFee` and `noPriorityFee` are never used on their own, they always passed to other functions along with `suggestedFees`. In `r.evaluateAndUpdatePathDetails`, in `r.buildPath`, which makes them logically a part of SuggestedFees structure.
So we can move these 2 fields inside SuggestedFees and avoid 2 extra params

saledjenic replied:
@vkjr yes, I was thinking about that initially, but`SuggestedFees` type carries the info about fees, how much is each fee, base fee, priority fee... but if there is a base fee or not and if there is a priority fee or not are more fields that are describing the chain behaviour. So the chain can be base-feeless and/or priority-feeless and if the chan has no base fee `baseFee` field in the `SuggestedFees` type will be `0`, the same logic for priority fee. So that's the reason why they are not there.

vkjr replied:
That makes sense.
If `noBaseFee` and `noPriorityFee` are used to update chain fields, can we just carry this information using SuggestedFess structure?
For example right now inside `SuggestedFees` function we are getting noBaseFee and noPriorityFee:
`noBaseFee, noPriorityFee, err = f.getGaslessParamsForAccount(ctx, chainID, address)`

And use them:
```
if noBaseFee {
	baseFee = big.NewInt(0)
}
```

But we can also use nil value assigned to `baseFee` that can indicate that the chain doesn't have  baseFee at all. That would result in:
```
if noBaseFee {
	baseFee = nil
}
```

And we won't have to pass 2 extra params everywhere. Wdyt?

saledjenic replied:
Yes, technically we could do that, but I wanted to follow safer solution, cause that one we would most likely require clients updates, cause I am not sure if clients are handling properly `null` values for those fields and most likely we would need many more lines in functional and unit tests to update.

friofry commented on rpc/chain/client.go:
```diff
@@ -655,6 +657,30 @@ func (c *ClientWithFallback) FeeHistory(ctx context.Context, blockCount uint64,
 }
 
 func (c *ClientWithFallback) EstimateGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error) {
+	if c.ChainID == walletCommon.StatusNetworkSepolia {
```
1) Can you move this part (`if chainID == StatusNetworkSepolia`) to a separate method.
I think this violates the Open/Closed Principle from SOLID (open for extension but closed for modification). It adds complexity without any benefits. Harder to read, maintain. 

2) `StatusNetworkSepoliaChainID` is also defined in `const.go`

saledjenic replied:
> StatusNetworkSepoliaChainID is also defined in const.go

One from wallet constants should be used, I can add a commit on top of with that change

> Can you move this part (if chainID == StatusNetworkSepolia) to a separate method.

@friofry  I can, but in that case this `EstimateGas` doesn't work for Status Chain, I mean, doesn't give the correct value.

friofry replied:
> doesn't give the correct value.

I guess it's fine. I think this business logic is not the matter of RPC connection wrapper

saledjenic replied:
@friofry new commit added for constants.

saledjenic replied:
@friofry it's not a quick thing to split that here, cause our processors are estimating gas via `GasEstimator` interface of go-ethereum, and I would need to update our fork to make it different, which I don't want, since we want to make changes on our side that will let us update easily to the latest go-ethereum implementation.

Need to think and check the code on how to handle that on our side, but that will definitely be a new PR.

saledjenic committed:
chore(api)_: removed constants from api common package and chain constants from wallet common are used instead

- Removed api `common` package since except in a few places the app is using chain constants from the wallet `common` package
- All references to chain IDs from removed package are updated

friofry reviewed (approved):
Looks good!

Hopefully, we can remove the hardcoded complex logic from client.go in the follow-up task
